test/trace-test.js

```diff
@@ -0,0 +1,135 @@
+
+    store.add(config, opts, function onAdd(err, tracer) {
+        assert.false(err, 'no error, idempotent add');
+        assert.true(tracer, 'tracer already added');
+        assert.equal(tracer, theTracer, 'tracer returned is same');
+    });
+
+    // remove twice, checking both times
+    store.remove(config, opts, function onRemove(err, tracer) {
+        assert.false(err, 'no error, idempotent remove');
+        assert.true(tracer, 'tracer removed');
+        assert.equal(tracer, theTracer, 'tracer returned is same');
+    });
+
+    store.remove(config, opts, function onRemove(err, tracer) {
+        assert.false(err, 'no error, idempotent remove');
+        assert.false(tracer, 'tracer already removed');
+    });
+});
+
+test('trace events get traced', function t(assert) {
+    var ringpop = new rpMock();
+    var store = new TracerStore(ringpop);
+    var config = core.resolveEventConfig(ringpop, 'membership.update');
+    var opts = {expiresIn: 50, sink: {type: 'log'}};
+    var tracer = store.add(config, opts);
+
+    assert.plan(5);
+    tracer.on('connectSink', function onTracer() {
+        // now test events on this thing
+        assert.equal(ringpop.journal.length, 0, 'journal starts empty');
+        ringpop.membership.emit('updated', 'foo');
+        assert.equal(ringpop.journal.length, 1, 'journal has 1 entry');
+        assert.equal(ringpop.journal[0], 'foo', 'journal added foo');
+
+        ringpop.membership.emit('updated', 'bar');
+        assert.equal(ringpop.journal.length, 2, 'journal has 2 entries');
+        assert.equal(ringpop.journal[1], 'bar', 'journal added bar');
+    });
+});
+
+test('trace events can renew, and expire', function t(assert) {
+    var ringpop = new rpMock();
+    var store = new TracerStore(ringpop);
+    var config = core.resolveEventConfig(ringpop, 'membership.update');
+    var opts = {expiresIn: 50, sink: {type: 'log'}};
+    var tracer = store.add(config, opts);
+
+    assert.plan(3);
+
+    // nextTick okay since log "connects" instantly
+    process.nextTick(function onTracer() {
+        // now test events on this thing
+        ringpop.membership.emit('updated', 'foo');
+        assert.equal(ringpop.journal.length, 1, 'journal has 1 entry');
+
+        // renew after 25, run @ 60, confirm event; this is racey...
+        setTimeout(store.add.bind(store, config, opts), 25);
+        setTimeout(function() {
+            ringpop.membership.emit('updated', 'bar');
+            assert.equal(ringpop.journal.length, 2, 'journal has 2 entries');
+        }, 60);
+
+        // let it expire, emit, confirm no log
+        setTimeout(function() {
+            ringpop.membership.emit('updated', 'dropped');
+            assert.equal(ringpop.journal.length, 2, 'journal unchanged');
+        }, 100);
+    });
+});
+
+// dumbest of mocks for our trace element
+function rpMock() {
+    var journal = [];
+    this.journal = journal;
+    this.logger = {
```

Comment:
Using logger calls to validate assertions seems error prone. While I understand that you're using the log tracer which writes to the log, any logging statements that are introduced will throw these tests off. Can you instead attach a listener to the send event of the LogTracer and journal that way?